internal/controllers/nodefeaturediscovery_reconciler_test.go

```diff
@@ -131,3 +138,70 @@ var _ = Describe("Reconcile", func() {
 		Entry("all components succeeded", nil, nil, nil, nil, nil, nil),
 	)
 })
+
+var _ = Describe("handleMaster", func() {
+	var (
+		ctrl           *gomock.Controller
+		clnt           *client.MockClient
+		mockDeployment *deployment.MockDeploymentAPI
+		nfdh           nodeFeatureDiscoveryHelperAPI
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		clnt = client.NewMockClient(ctrl)
+		mockDeployment = deployment.NewMockDeploymentAPI(ctrl)
+
+		nfdh = newNodeFeatureDiscoveryHelperAPI(clnt, mockDeployment, scheme)
+	})
+
+	ctx := context.Background()
+
+	It("new deployment", func() {
```

Comment:
nit: not a blocker, but a note these ginkgo descriptions could be a bit more verbose/descriptive, following the ginkgo concept of more expressive and easy-to-read tests

```suggestion
	It("should create new nfd-master deployment if it does not exist", func() {
```

WDYT?

Reply:
sure, fixed